packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,239 @@
+      let initialNextLandmark = nextLandmark;
+      do {
+        // If something was previously focused in the next landmark, then return focus to it
+        if (nextLandmark.lastFocused) {
+          let lastFocused = nextLandmark.lastFocused;
+          if (document.body.contains(lastFocused)) {
+            lastFocused.focus();
+            return;
+          }
+        }
+
+        // Otherwise, focus the first focusable element in the next landmark
+        let leadingSentinal = nextLandmark.ref.current.previousSibling || nextLandmark.ref.current.parentElement;
+        // If we want to add a scope, this is what we're thinking:
+        // let trailingSentinal = nextLandmark.ref.current.nextSibling;
+        // let scope = [leadingSentinal, trailingSentinal];
+        let walker = getFocusableTreeWalker(nextLandmark.ref.current, {tabbable: true});
+        let nextNode = walker.nextNode() as HTMLElement;
+        if (!nextNode) {
+          walker.currentNode = leadingSentinal;
+          nextNode = walker.nextNode() as HTMLElement;
+        }
+        while (nextNode && this.closestLandmark(nextNode) !== nextLandmark) {
+          nextNode = walker.nextNode() as HTMLElement;
+        }
+        if (document.body.contains(nextNode)) {
+          nextNode.focus();
+          return;
+        }
+        
+        nextLandmark = this.getNextLandmark(nextLandmark.ref.current, {backward});
+      } while (nextLandmark !== initialNextLandmark);
+     
+    }
+  }
+
+  // Only the closest landmark cares about focus inside of it
+  public focusinHandler(e: FocusEvent) {
+    let currentLandmark = this.closestLandmark(e.target as HTMLElement);
+    this.landmarks = this.landmarks.map((landmark) => ({...landmark, ...(landmark === currentLandmark && {lastFocused: e.target as HTMLElement})}));
+  }
+}
+
+document.addEventListener('keydown', LandmarkManager.getInstance().f6Handler.bind(LandmarkManager.getInstance()), {capture: true});
+document.addEventListener('focusin', LandmarkManager.getInstance().focusinHandler.bind(LandmarkManager.getInstance()), {capture: true});
+
+/**
+ * Provides landmark navigation in an application. Call this with a role and label to register a landmark navigable with F6.
+ * @param props - Props for the landmark.
+ * @param ref - Ref to the landmark.
+ */
+export function useLandmark(props: AriaLandmarkProps, ref: MutableRefObject<HTMLElement>): LandmarkAria {
+  const {
+    role,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby
+  } = props;
+  let manager = LandmarkManager.getInstance();
+  let label = ariaLabel || ariaLabelledby;
+
+  useLayoutEffect(() => {
+    manager.addLandmark({ref, role, label});
+    
+    return () => {
+      manager.removeLandmark(ref);
+    };
+  }, []);
+
+  useEffect(() => {
+    manager.updateLandmark({ref, label, role});
+  }, [label, ref, role]);
+
+  // let everything through? or only return role + labelling?
```

Comment:
Open question

Reply:
I think it depends if we primarily use it in other hooks or if it's used in react spectrum components primarily

My guess is hooks, and it should only return the role, not the labelling, because that'll be coming from the useLabel hook most likely, not on props.

@devongovett do you have opinions on this?

Reply:
just weighing in, +1 to role only

Reply:
Update to return just `role` and `tabIndex: -1`